pkg/common/datastore/datastore.go

```diff
@@ -111,6 +111,28 @@ func (a *ApplianceDatum) GetBladeDatumById(ctx context.Context, bladeId string)
 	return blade, nil
 }
 
+// Verify if the blade exists using the ipAddress
+func (c *DataStore) CheckBladeExist(IpAddress string) (*string, bool) {
+	for _, appliance := range c.ApplianceData {
+		for bladeId, blade := range appliance.BladeData {
+			if blade.Credentials.IpAddress == IpAddress {
+				return &bladeId, true
+			}
+		}
+	}
+	return nil, false
+}
+
+// Verify if the host exists using the ipAddress
+func (c *DataStore) CheckHostExist(IpAddress string) (*string, bool) {
+	for hostId, host := range c.HostData {
+		if host.Credentials.IpAddress == IpAddress {
+			return &hostId, true
+		}
+	}
+	return nil, false
+}
+
```

Comment:
A couple things with these 2 functions:
1.) Please move them so they are grouped with the other `func (c *DataStore)` functions.
I make great efforts to try and keep the files organized by putting related code next to each other in the file (at least by struct type).   While some people may not agree with this, I think it makes the code easier to read and more professional looking.
2.) Let's try and keep a similar naming convention to what we use elsewhere in the code.  I'm thinking `GetBladeDatumByIp()`, for example.